libs/sdk-model/src/accessControl/index.ts

```diff
@@ -44,12 +44,40 @@ export const isUserGroupAccess = (obj: unknown): obj is IUserGroupAccess => {
     return !isEmpty(obj) && (obj as IUserGroupAccess).type === "group";
 };
 
+/**
+ * User access specification with granular permissions.
+ *
+ * @alpha
+ */
+export type IGranularUserAccess = IUserAccess & IGranularAccessGrantee;
+
+/**
+ * User group access specification with granular permissions.
+ *
+ * @alpha
+ */
+export type IGranularUserGroupAccess = IUserGroupAccess & IGranularAccessGrantee;
+
+/**
+ * Tests whether the provided object is an instance of {@link IGranularUserAccess} or {@link IGranularUserGroupAccess}.
+ *
+ * @param obj - object to test
+ * @alpha
+ */
+export const isGranularAccess = (obj: unknown): obj is IGranularUserAccess | IGranularUserGroupAccess => {
+    return !isEmpty(obj) && (obj as IGranularUserAccess | IGranularUserGroupAccess).permissions !== undefined;
```

Comment:
is this typeguard safe? check just for `!== undefined` seems potentially risky

Reply:
I will rather use `has(obj, property)`

Reply:
The main point is that the check is probably too vague. (similar to the other type guards in this file) Imagine `{ permissions: true }` or `{ permissions: {} }` both will pass but are totally different types.